clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/LoadBalancersSpec.java

```diff
@@ -0,0 +1,254 @@
+  private final String ECS_ACCOUNT_NAME = "ecs-account";
+  private final String TEST_REGION = "us-west-2";
+
+  @Test
+  public void getLoadBalancersTest() {
+    // given
+    ProviderCache ecsCache = providerRegistry.getProviderCache(EcsProvider.NAME);
+    String testNamespace = Namespace.LOAD_BALANCERS.ns;
+    String loadBalancerKey =
+        com.netflix.spinnaker.clouddriver.aws.data.Keys.getLoadBalancerKey(
+            testNamespace, "*", TEST_REGION, "*", "*");
+
+    Set<String> keys = new HashSet<>();
+    keys.add(
+        "aws:targetGroups:my-aws-devel-acct:us-west-2:spinnaker-ecs-demo-artifacts-tg:ip:vpc-07daae48bf98a8fd8");
+
+    List<String> securityGroups = new ArrayList<>();
+    securityGroups.add("test-security");
+
+    Map<String, Collection<String>> relationships = new HashMap<>();
+    relationships.put("targetGroups", keys);
+
+    Map<String, Object> attributes = new HashMap<>();
+    attributes.put("account", ECS_ACCOUNT_NAME);
+    attributes.put("region", TEST_REGION);
+    attributes.put("name", ECS_ACCOUNT_NAME);
+    attributes.put("vpcId", "vpc-123");
+    attributes.put("loadBalancerType", "test-type");
+    attributes.put("securityGroups", securityGroups);
+    attributes.put("targetGroups", "test-target");
+    attributes.put("loadBalancerName", "testLB");
+
+    DefaultCacheResult testResult =
+        buildCacheResultForLB(attributes, testNamespace, loadBalancerKey, relationships);
+    ecsCache.addCacheResult("TestAgent", Collections.singletonList(testNamespace), testResult);
+
+    String url = getTestUrl("/ecs/loadBalancers");
+
+    when(mockEcsAccountMapper.fromAwsAccountNameToEcsAccountName("*")).thenReturn(ECS_ACCOUNT_NAME);
+
+    // when
+    Response response =
+        get(url).then().statusCode(200).contentType(ContentType.JSON).extract().response();
+
+    String responseStr = response.asString();
+    assertTrue(responseStr.contains(ECS_ACCOUNT_NAME));
+    assertTrue(responseStr.contains(TEST_REGION));
+    assertTrue(responseStr.contains("spinnaker-ecs-demo-artifacts-tg"));
+    assertTrue(responseStr.contains("testLB"));
+    assertTrue(responseStr.contains("testLB"));
+    assertTrue(responseStr.contains("vpc-123"));
+    assertTrue(responseStr.contains("test-security"));
+  }
+
+  @Test
+  public void getLoadBalancersForApplicationTest() {
+    // given
+    ProviderCache ecsCache = providerRegistry.getProviderCache(EcsProvider.NAME);
+    String testNamespaceForLB = Namespace.LOAD_BALANCERS.ns;
+    String loadBalancerKey =
+        com.netflix.spinnaker.clouddriver.aws.data.Keys.getLoadBalancerKey(
+            testNamespaceForLB, "*", TEST_REGION, "*", "*");
+
+    Map<String, Collection<String>> relationships = new HashMap<>();
+    relationships.put("loadBalancers", Arrays.asList(loadBalancerKey));
+    relationships.put(
+        "targetGroups",
+        Arrays.asList(
+            "aws:targetGroups:aws-account:us-west-2:spinnaker-ecs-demo-tg:ip:vpc-07daae48bf98a8fd8"));
```

Comment:
since we later assert on this string (and the target group arn on line 151), can we define and use it as a variable instead?

Reply:
Resolved.